onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/resize_op_builder.cc

```diff
@@ -224,14 +226,16 @@ bool ResizeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers
       LOGS_DEFAULT(VERBOSE) << "Input sizes of Resize must be known";
       return false;
     }
-
+    bool use_nchw = false;
+    // haven't a good solution to check layout when scale is 1.0F
     // We want to check if the scales or sizes are not trying to resize on N/C channels here
     if (inputs.size() == 3) {  // we are using scales
       const auto& scales_tensor = *initializers.at(inputs[2].node_arg.Name());
-      Initializer unpacked_tensor(scales_tensor);
+      Initializer const unpacked_tensor(scales_tensor);
       auto scales_data = unpacked_tensor.DataAsSpan<float>();
-      float scale_n = scales_data[0];
-      float scale_c = IsNodeLayoutNHWC(node_unit) ? scales_data[3] : scales_data[1];
+      use_nchw = scales_data[1] == 1.0F;
+      float const scale_n = scales_data[0];
+      float const scale_c = !use_nchw ? scales_data[3] : scales_data[1];
```

Comment:
nit: adding the '!' to the condition is an unnecessary complication

Reply:
Sure.